Xwt.XamMac/Xwt.Mac/PopoverBackend.cs

```diff
@@ -36,139 +38,251 @@ namespace Xwt.Mac
 {
 	public class PopoverBackend : IPopoverBackend
 	{
+		public Popover Frontend { get; private set; }
+		public ApplicationContext ApplicationContext { get; set; }
+		public IPopoverEventSink EventSink { get; set; }
+		internal bool EnableCloseEvent { get; private set; }
 		NSPopover popover;
-		public event EventHandler Closed;
 
-		class FactoryViewController : NSViewController
+		class FactoryViewController : NSViewController, INSPopoverDelegate
 		{
-			Xwt.Widget child;
-			NSView view;
+			// Always retain the object in a cache until it closes. This
+			// guarantees that neither the NSPopover nor the native FactoryViewController
+			// can be GC'ed until after it has closed.
+			static readonly HashSet<FactoryViewController> Cache = new HashSet<FactoryViewController> ();
+
+			public Widget Child { get; private set; }
+			public PopoverBackend Backend { get; private set; }
 			public CGColor BackgroundColor { get; set; }
+			public ViewBackend ChildBackend { get; private set; }
+			public NSView NativeChild { get { return ChildBackend?.Widget; } }
+			bool shown;
+			NSPopover parent;
 
-			public FactoryViewController (Xwt.Widget child) : base (null, null)
+			public FactoryViewController (PopoverBackend backend, Widget child, NSPopover parentPopover) : base (null, null)
 			{
-				this.child = child;
+				Child = child;
+				ChildBackend = Toolkit.GetBackend (Child) as ViewBackend;
+				Backend = backend;
+				parent = parentPopover;
+				Cache.Add (this);
 			}
 
-			// Called when created from unmanaged code
-			public FactoryViewController (IntPtr handle) : base (handle)
+			public override void LoadView ()
 			{
+				View = new ContainerView (this);
+				NativeChild.RemoveFromSuperview ();
+				View.AddSubview (NativeChild);
+
+				WidgetSpacing padding = 0;
+				if (Backend != null)
+					padding = Backend.Frontend.Padding;
+				View.AddConstraints (new NSLayoutConstraint [] {
```

Comment:
Should likely disable the NativeChild's translatesAutoresizingMaskIntoConstraints property